### PR TITLE
Austenem/CAT-834 restyle attribution banner

### PR DESCRIPTION
**Pull request by austenem**
## Summary

This updates the banner above the "Contributors" and "Attribution" sections from an alert to a styled paper component.

## Design Documentation/Original Tickets

[CAT-834 Jira ticket](https://hms-dbmi.atlassian.net/browse/CAT-834?atlOrigin=eyJpIjoiYzQxNDA1NTAzNjZiNDNhYWJlYjdhYTgwYjBiNjQ4Y2MiLCJwIjoiaiJ9)

[Figma mockups](https://www.figma.com/design/awtd3Klx2hka54jyekr9pV/Detail-Pages?node-id=5258-20813&m=dev)

## Testing

Manually tested by visually comparing both banners to designs.

Dataset with an attribution banner: http://localhost:5001/browse/dataset/f805a2fd3549d66a5ffee47b00add18c#attribution
Dataset with a contributors banner: http://localhost:5001/browse/dataset/57b475b8ad39fa2cf07ede0cf192b2f6#contributors

## Screenshots/Video

Screenshot 2024-08-14 at 11 11 46 AM
Screenshot 2024-08-14 at 11 15 36 AM

## Checklist

- [X] Code follows the project's coding standards
  - [X] Lint checks pass locally
  - [X] New `CHANGELOG-your-feature-name-here.md` is present in the root directory, describing the change(s) in full sentences.
- [ ] Unit tests covering the new feature have been added
- [X] All existing tests pass
- [X] Any relevant documentation in JIRA/Confluence has been updated to reflect the new feature
- [X] Any new functionalities have appropriate analytics functionalities added

### CODE REVIEW

File: context/app/static/js/shared-styles/alerts/InfoAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import Stack from '@mui/material/Stack';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+
+import { InfoIcon } from 'js/shared-styles/icons';
+
+function InfoAlert({ children }: { children: React.ReactNode }) {
+  return (
+    <Stack component={Paper} p={2} spacing={2} marginBottom={1.25}>
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/alerts/InfoAlert.tsx:11]: 1.25 margin evaluates to 10px, I believe. This isn't a value I see often in the design. Did this come from the design?

File: context/app/static/js/shared-styles/alerts/InfoAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+      <Stack direction="row" spacing={2}>
+        <InfoIcon color="primary" fontSize="1.5rem" />
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/alerts/InfoAlert.tsx:13]: Could we possibly reuse the `iconSymbolStatusMap` used for the `StatusIcon` to make this a bit more reusable? I believe we have like components of different severities that could be replaced with this component.

File: context/app/static/js/shared-styles/alerts/InfoAlert.tsx (hunk 1 above)
Comment (john-conroy) [inline context/app/static/js/shared-styles/alerts/InfoAlert.tsx:11]: Why the two stacks? The outer stack only has a single child.

File: context/app/static/js/shared-styles/alerts/InfoAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+
+import Stack from '@mui/material/Stack';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+
+import { InfoIcon } from 'js/shared-styles/icons';
+
+function InfoAlert({ children }: { children: React.ReactNode }) {
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/alerts/InfoAlert.tsx:9]: Since this isn't an alert perhaps a different name?

**Review by john-conroy — CHANGES_REQUESTED** [review]
Thanks! I don't want to hold this up so let's not worry about replacing the similar components for now, but let's file a ticket.

File: context/app/static/js/shared-styles/alerts/InfoAlert.tsx (hunk 1 above)
Comment (austenem) [reply context/app/static/js/shared-styles/alerts/InfoAlert.tsx:11]: The designs actually specify that this should be 8px, so I'll update to `marginBottom={1}`.

**Comment by austenem** [comment]
If there's a better way to generalize the `iconSymbolStatusMap` and `ColoredStatusIconProps`, I'd be open to it - these seem like they could be put into a better location and renamed (because they are now being used in multiple components spread across features), but I'm not sure where. Alternatively, if there's a way to override the `ColoredStatusIcon` styles in order to just use that styled component, that could be helpful.

**Comment by john-conroy** [comment]
You could introduce something like the below into `shared-styles/icons` and make it work for both this case and the `StatusIcon`.

```
import React from 'react';
import { InfoRounded, ErrorRounded, WarningRounded, CheckCircleRounded } from '@mui/icons-material';
import SvgIcon, { SvgIconProps } from '@mui/material/SvgIcon';

type IconStatus = 'info' | 'success' | 'warning' | 'error';

export interface ColoredStatusIconProps extends SvgIconProps {
  $iconStatus: IconStatus;
}

export const iconSymbolStatusMap: Record<IconStatus, React.ComponentType<SvgIconProps>> = {
  info: InfoRounded,
  success: CheckCircleRounded,
  warning: WarningRounded,
  error: ErrorRounded,
};

function SeverityIcon({ status, ...svgIconProps }: Partial<SvgIconProps> & { status: IconStatus }) {
  return <SvgIcon color={status} component={iconSymbolStatusMap[status]} {...svgIconProps} />;
}

export default SeverityIcon;
```

File: context/app/static/js/components/detailPage/StatusIcon/ColoredStatusIcon/style.ts [hunk 4]
```diff
@@ -1,20 +1,7 @@
...
-
-export const iconSymbolStatusMap: {
-  [key in ColoredStatusIconProps['$iconStatus']]: React.ComponentType<SvgIconProps>;
-} = {
-  info: InfoRounded,
-  success: CheckCircleRounded,
-  warning: WarningRounded,
-  error: ErrorRounded,
-};
+import { ColoredStatusIconProps } from 'js/shared-styles/icons/SeverityIcon';
```
Comment (john-conroy) [inline context/app/static/js/components/detailPage/StatusIcon/ColoredStatusIcon/style.ts:4]: We should be able to get rid of this file, use `SeverityIcon` directly in `StatusIcon` and style it there using the `sx` prop to apply its unique styling there.

File: context/app/static/js/shared-styles/panels/IconPanel/style.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { SvgIconProps } from '@mui/material/SvgIcon';
+import { styled } from '@mui/material/styles';
+
+export const getStyledIcon = (IconComponent: React.ComponentType<SvgIconProps>) =>
+  styled(IconComponent)(({ theme }) => ({
+    color: theme.palette.primary.main,
+    fontSize: '1.5rem',
+  }));
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/panels/IconPanel/style.ts:8]: We should be able to get rid of this and apply styles to `SeverityIcon` directly.

File: context/app/static/js/shared-styles/panels/IconPanel/IconPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import React, { PropsWithChildren } from 'react';
+
+import Stack from '@mui/material/Stack';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+
+import SeverityIcon, { ColoredStatusIconProps } from 'js/shared-styles/icons/SeverityIcon';
+
+interface IconPanelProps extends PropsWithChildren {
+  $iconStatus: ColoredStatusIconProps['$iconStatus'];
```
Comment (john-conroy) [inline context/app/static/js/shared-styles/panels/IconPanel/IconPanel.tsx:10]: We start props with `$` if they are only used in styling. Can we reuse the `IconStatus` type in `SeverityIcon` here? 

```
{status: IconStatus}
```

**Review by john-conroy — APPROVED** [review]
Looks good, thanks!

**Pull request merged**